launchflow/workflows/tf/resources/gcp_backend_bucket/main.tf

```diff
@@ -0,0 +1,114 @@
+provider "google" {
+  project = var.gcp_project_id
+  region  = var.gcp_region
+}
+
+resource "google_storage_bucket_iam_member" "bucket_iam" {
+  bucket = google_storage_bucket.bucket.name
+  role   = "roles/storage.objectViewer"
+  member = "allUsers"
+}
+
+resource "google_storage_bucket" "bucket" {
+  name                        = var.resource_id
+  project                     = var.gcp_project_id
+  location                    = var.location
+  force_destroy               = var.force_destroy
+  uniform_bucket_level_access = true
+  website {
+    main_page_suffix = var.main_page_suffix
+    not_found_page   = var.not_found_page
+  }
+}
+
+resource "google_storage_bucket_object" "index_html" {
```

Comment:
If you create this resource, then deploy to it I assume this file would get overritten?

What happens if they change some config on the backend bucket then? Like update the 404 path or something. I _think_ it would cause the contents of the deployed index to get overwritten with the hello world contents. You will probably need to add a life cycle ignore

Reply:
yeah Im gonna polish this up tomorrow. Current plan is to redo this so each deployment gets its own directory in the bucket and then we'll switch the directory we serve. Thinking it will be a quick way to support rollbacks and whatnot